src/lib.rs

```diff
@@ -161,6 +161,18 @@ pub enum BlockBasedIndexType {
     TwoLevelIndexSearch,
 }
 
+/// Used by BlockBasedOptions::set_data_block_index_type.
+pub enum DataBlockIndexType {
```

Comment:
```suggestion
#[repr(C)]
pub enum DataBlockIndexType {
```

Reply:
Would this let me avoid doing [this](https://github.com/rust-rocksdb/rust-rocksdb/pull/358/files#diff-915d9d8a57469f27ff59479e2b15dba1R208)?

Reply:
no, but I think the default repr for enums is `isize` https://github.com/rust-lang/rust/issues/24290,
so even though in practice it doesn't cause any trouble, we should be the safe side with `[repr(C)]` or `#[repr(i32)]`